taskcluster/client.py

```diff
@@ -536,9 +536,19 @@ def _makeHttpRequest(self, method, route, payload):
 
             # Try to load JSON
             try:
+                if response.headers or headers in response:
+                    if 'Content-Type' in response.headers and \
+                            'application/json' not in response.headers['Content-Type']:
```

Comment:
Per the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) and [HTTP spec](https://tools.ietf.org/html/rfc7231#section-3.1.1.5), we know that the format for content-type headers are going to be something like `application/json` or `application-json; charset=utf-8`.  I think we'd have a more robust solution here if we split the header value by `;` character, and check that the trimmed first index is exactly `application/json`.  This way we know for sure that when we're parsing the response as JSON, that we're expecting a JSON value.

In an ideal world, we'd use the `charset` parameter if present to pass to the JSON parser to interpret the body correctly, but I think we make enough assumptions about UTF-8 being used elsewhere that this isn't important.

Reply:
I see, thats a good idea. I'll modify it according to your suggestion. thanks!!